index.d.ts

```diff
@@ -1,5 +1,9 @@
 declare module 'react-string-replace' {
-  function reactStringReplace(text: React.ReactNode, regex: RegExp, cb: (match: string, index: number) => string | JSX.Element): JSX.Element;
+  function reactStringReplace(
+    text?: string | React.ReactNodeArray, 
```

Comment:
Intention is:
Initially you would pass a `string`. Also you can pass the result of of the function again, which would be `React.ReactNodeArray`.